sonar-ballerina-plugin/build.gradle

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.jar.JarInputStream
+
+plugins {
+    id 'java'
+    id 'java-library'
+    id 'checkstyle'
+    id "com.github.spotbugs"
+    id "de.undercouch.download"
+    id "com.github.johnrengelman.shadow"
```

Comment:
@Xenowa Are we using this to create an Uber jar?

Reply:
Yes, 

SonarQube language plugins are designed to work in a standalone manner in any environment where the analysis was triggered from.  Due to this, if there are dependencies like `Gson` that is needed to parse JSON files in a language plugin, they need to be packaged with the plugin JAR. 

If instead a thin JAR is placed in the 'extensions' directory of the SonarQube installation, it will not start up due to the missing dependencies in the placed JAR.

For this reason the shadow plugin dependency has been included.

Reply:
If its  for Uber jar creation can't we use something like [this](https://github.com/ballerina-platform/ballerina-update-tool/blob/master/build.gradle#L39) ? We had requirement for update-tool to create a uber jar as it should work stand alone and was able to create one without an extra dependency

Reply:
> If its for Uber jar creation can't we use something like [this](https://github.com/ballerina-platform/ballerina-update-tool/blob/master/build.gradle#L39) ? We had requirement for update-tool to create a uber jar as it should work stand alone and was able to create one without an extra dependency

Changes have been applied with #[8594038](https://github.com/ballerina-platform/sonar-ballerina/blob/85940385e30f31ecd4cf1635fc456153337ed133/sonar-ballerina-plugin/build.gradle#L158)